db/queries/statuses/queries.py

```diff
@@ -81,58 +81,70 @@ def update_form_statuses(
+        # when submitted with no answer
+        case None:
+            return True
+        # default case when there is an answer
+        case _:
+            return True
+
+
+def _determine_answer_status_for_fields(fields_in_question) -> list[bool]:
+    answer_found_list = [_is_field_answered(field) for field in fields_in_question]
+    return answer_found_list
+
+
+def _determine_question_status_from_answers(answer_found_list: list[bool]) -> str:
+
+    # If we found no answers
+    if not answer_found_list or len(answer_found_list) == 0:
+        return "NOT_STARTED"
+    # If all answers are given
+    if all(answer_found_list):
+        return "COMPLETED"
+    # If some answers are given
+    elif any(answer_found_list):
+        return "IN_PROGRESS"
+    # If no answers are given
+    else:
+        return "NOT_STARTED"
+
+
+def update_question_statuses(stored_form_json):
     """
     Updates the question statuses of each question if a value is present
 
     Args:
         application_id: The application id
     """
-    stored_forms = get_application(
-        application_id, as_json=False, include_forms=True
-    ).forms
+    for question_page in stored_form_json:
+        question_page["status"] = question_page.get("status", "NOT_STARTED")
 
-    for stored_form in stored_forms:
-        if stored_form.name == form_name:
-            for question_page in stored_form.json:
-                question_page["status"] = question_page.get("status", "NOT_STARTED")
-                if question_page["status"] == "SUBMITTED":
-                    break
-
-                def is_field_answered(field):
-                    answer_or_not_specified = field.get("answer")
-                    match answer_or_not_specified:  # noqa
-                        case "":
-                            return False
-                        case []:  # noqa (E211)
-                            return False
-                        # optional questions return None (not string)
-                        # when submitted with no answer
-                        case None:
-                            return True
-                        # default case when there is an answer
-                        case _:
-                            return True
-
-                answer_found_list = [
-                    is_field_answered(field) for field in question_page["fields"]
-                ]
-                # If all answers are given
-                if all(answer_found_list):
-                    question_page["status"] = "COMPLETED"
-                # If some answers are given
-                elif not all([not found_answer for found_answer in answer_found_list]):
-                    question_page["status"] = "IN_PROGRESS"
-                # If no answers are given
-                else:
-                    question_page["status"] = "NOT_STARTED"
-    db.session.commit()
+        # doesn't look like we ever set a question_page status to be SUBMITTED
```

Comment:
Could we remove the comments & redundant code, although you did mention this is a draft so maybe that’s due to be done?